### PR TITLE
HourlyWind Enhancement

### PR DESCRIPTION
**Pull request by Patowhiz**

### CODE REVIEW

File: ClimsoftVer4/ClimsoftVer4/ucrHourlyWind.vb [hunk 1]
```diff
@@ -393,21 +400,85 @@ Public Class ucrHourlyWind
...
+            'if selectedDate  is earlier than todayDate (<0)  then its a valid date for data entry
+            'if it is same time (0) or later than (>0) then its invalid, disable control
+            Me.Enabled = If(Date.Compare(selectedDate, todayDate) < 0, True, False)
         Else
-            'if it is same time (0) or later than (>0) disable control
             Me.Enabled = False
         End If
     End Sub
 
+    Public Sub UploadAllRecords()
```
Comment (maxwellfundi) [inline ClimsoftVer4/ClimsoftVer4/ucrHourlyWind.vb:419]: @Patowhiz  Why do we have this code here? Are you testing the upload? Seems like this is frmDaily code

Comment (Patowhiz) [reply ClimsoftVer4/ClimsoftVer4/ucrHourlyWind.vb:419]: I'm not done with this code yet. I just copied it from frmDaily to prepare this form for the upload code

Comment (maxwellfundi) [reply ClimsoftVer4/ClimsoftVer4/ucrHourlyWind.vb:419]: Oaky great, then i think you can merge this too

Comment (Patowhiz) [reply ClimsoftVer4/ClimsoftVer4/ucrHourlyWind.vb]: @dannyparsons this is how I'm handling the change of focus on enter to the next control

```
 ''' <summary>
    ''' sets key down listeners for the form controls
    ''' </summary>
    Private Sub SetControlsKeyDownListners()
        Dim synopticControl As ucrSynopticRA1

        For Each formCtr As Control In Me.Controls
            If TypeOf formCtr Is ucrBaseDataLink Then
                If TypeOf formCtr Is ucrSynopticRA1 Then
                    'for ucrSynopticRA1 controls
                    synopticControl = DirectCast(formCtr, ucrSynopticRA1)
                    For Each synopCtr As Control In synopticControl.Controls
                        If TypeOf synopCtr Is ucrValueFlagPeriod Then
                            AddHandler DirectCast(synopCtr, ucrValueFlagPeriod).evtKeyDown, AddressOf GoToNextControl
                        End If
                    Next
                Else
                    'for the other base controls e.g station, year, month, day, hour slectors
                    AddHandler DirectCast(formCtr, ucrBaseDataLink).evtKeyDown, AddressOf GoToNextControl
                End If
            End If
        Next
    End Sub

    ''' <summary>
    ''' determines the next control to get focus on enter key  
    ''' </summary>
    ''' <param name="sender"></param>
    ''' <param name="e"></param>
    Private Sub GoToNextControl(sender As Object, e As KeyEventArgs)
        If e.KeyCode = Keys.Enter Then
            Dim bGoToNextControl As Boolean = False

            'do validations to determine whether to go to next control
            If TypeOf sender Is ucrValueFlagPeriod Then
                If DirectCast(sender, ucrValueFlagPeriod).PreValidateValue() Then
                    bGoToNextControl = True
                End If
            ElseIf TypeOf sender Is ucrBaseDataLink Then
                'for the other base controls e.g station, year, month, day, hour slectors
                If DirectCast(sender, ucrBaseDataLink).ValidateValue() Then
                    bGoToNextControl = True
                End If
            End If

            If bGoToNextControl Then
                Me.SelectNextControl(sender, True, True, True, True)
            End If

            'to handle the "noise"
            e.SuppressKeyPress = True
        End If
    End Sub
```

**Pull request merged**